File: app/src/main/java/org/dhis2/utils/RulesActionCallbacks.java
package org.dhis2.utils;

import android.support.annotation.NonNull;
import android.support.annotation.Nullable;

import org.hisp.dhis.rules.models.RuleActionShowError;

/**
 * QUADRAM. Created by ppajuelo on 13/06/2018.
 */

public interface RulesActionCallbacks {


    void setShowError(@NonNull RuleActionShowError showError);

    void unsupportedRuleAction();

    void save(@NonNull String uid, @Nullable String value);

    void setDisplayKeyValue(String label, String value);

    void sethideSection(String sectionUid);

    void setMessageOnComplete(String content, boolean canComplete);

    void setHideProgramStage(String programStageUid);
}

File: app/src/main/java/org/dhis2/utils/RulesActionCallbacks.kt
package org.dhis2.utils

import org.dhis2.form.model.FieldUiModel
import org.hisp.dhis.rules.models.RuleActionShowError

interface RulesActionCallbacks {

    fun setShowError(showError: RuleActionShowError, model: FieldUiModel?)

    fun unsupportedRuleAction()

    fun save(uid: String, value: String?)

    fun setMessageOnComplete(content: String, canComplete: Boolean)

    fun setHideProgramStage(programStageUid: String)

    fun setOptionToHide(optionUid: String, field: String)

    fun setOptionGroupToHide(optionGroupUid: String, toHide: Boolean, field: String)
}
